packages/react-datetime/src/components/CalendarMonth/CalendarMonth.tsx

```diff
@@ -6,42 +6,41 @@ import ArrowLeftIcon from '@patternfly/react-icons/dist/js/icons/arrow-left-icon
 import ArrowRightIcon from '@patternfly/react-icons/dist/js/icons/arrow-right-icon';
 import { css } from '@patternfly/react-styles';
 import styles from '@patternfly/react-styles/css/components/CalendarMonth/calendar-month';
-// https://date-fns.org/v2.16.1/docs/format
-import { format } from 'date-fns';
-import { Locales, Locale } from '../../helpers';
 import commonStyles from '@patternfly/react-styles/css/base/patternfly-common';
 import { getUniqueId } from '@patternfly/react-core/dist/js/helpers/util';
 
 export interface CalendarProps extends Omit<React.HTMLProps<HTMLDivElement>, 'onChange'> {
   /** Month/year to base other dates around */
   date: Date;
-  /** How to format months in dropdown according to date-fns */
-  monthFormat?: string;
-  /** How to format days in header according to date-fns */
-  dayFormat?: string;
-  /** How to format days in header for screen readers according to date-fns */
-  longDayFormat?: string;
-  /** How to format days in buttons according to date-fns */
-  buttonFormat?: string;
+  /** How to format months in Select */
+  monthFormat?: (date: Date) => React.ReactNode;
+  /** How to format week days in header */
+  weekdayFormat?: (date: Date) => React.ReactNode;
+  /** How to format days in header for screen readers */
+  longWeekdayFormat?: (date: Date) => React.ReactNode;
+  /** How to format days in buttons in table cells */
+  buttonFormat?: (date: Date) => React.ReactNode;
+  /** If using the default formatters which locale to use. Undefined defaults to current locale. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl#Locale_identification_and_negotiation */
+  locale?: string;
+  /** Day of week that starts the week. 0 is Sunday, 6 is Saturday. */
+  weekStart?: 0 | 1 | 2 | 3 | 4 | 5 | 6;
```

Comment:
Might be nicer to see `SUN | MON | TUE | WED | THU | FRI | SAT` for less mental mapping

Reply:
I've added a `Weekday` enum.